unison-cli/src/Unison/Codebase/Editor/HandleInput/LoopState.hs

```diff
@@ -80,3 +107,12 @@ currentPath = currentPathStack . to Nel.head
 
 loopState0 :: Branch m -> Path.Absolute -> LoopState m v
 loopState0 b p = LoopState b b (pure p) Nothing Nothing Nothing []
+
+respond :: MonadCommand n m i v => Output v -> n ()
```

Comment:
Moved these to the definition of `Action` to avoid cyclic imports.

We should really consider a lighter-weight class just for writing output to console.